### PR TITLE
Add support for protobuf generated PHP code.

### PR DESCRIPTION
**Pull request by stephenthedev**
## Description
Protobuf can generate PHP code, so collapsing in PRs helps legibility of those PRs.

## Checklist:

- [x] **I am adding new or changing current functionality**
  
  - [x] I have added or updated the tests for the new or changed functionality.

### CODE REVIEW

File: lib/linguist/generated.rb [hunk 1]
```diff
@@ -314,7 +314,7 @@ def generated_protocol_buffer?
       return false unless PROTOBUF_EXTENSIONS.include?(extname)
       return false unless lines.count > 1
 
-      return lines[0].include?("Generated by the protocol buffer compiler.  DO NOT EDIT!")
+      return lines.first(6).any? { |l| l.include?("Generated by the protocol buffer compiler.  DO NOT EDIT!") }
```
Comment (pchaigno) [inline lib/linguist/generated.rb]: I'm having trouble finding examples where this warning statement is further down than the 3rd line. Maybe change to the following?
```suggestion
      return lines.first(3).any? { |l| l.include?("Generated by the protocol buffer compiler.  DO NOT EDIT!") }
```

**Comment by raghunat** [comment]
@pchaigno updated! I think in previous similar PRs they used 6, so I just decided why not :)

**Comment by stephenthedev** [comment]
@lildude  thoughts? :-)

**Comment by stephenthedev** [comment]
@pchaigno @lildude  is there anything you need from me to get this through? My teams are reviewing lots of protobuf php code and this would make 2 cents better :-) Thanks!

**Comment by stephenthedev** [comment]
🥳 Thanks!

**Pull request merged**